orangewidget/utils/buttons.py

```diff
@@ -104,32 +81,88 @@ def sizeHint(self):
         icon = self.icon()
         if not icon.isNull():
             iconsize = icon.actualSize(iconsize)
-        return iconsize
+        margins = self.contentsMargins()
+        return QSize(
+            iconsize.width() + margins.left() + margins.right(),
+            iconsize.height() + margins.top() + margins.bottom()
+        )
 
     def minimumSizeHint(self):
         # reimplemented
         return self.sizeHint()
 
-    def paintEvent(self, event):
-        painter = QStylePainter(self)
-        option = QStyleOptionButton()
+    def initStyleOption(self, option: QStyleOptionButton):
         option.initFrom(self)
-        option.text = ""
-        option.icon = self.icon()
+        option.features |= QStyleOptionButton.Flat
         option.iconSize = self.iconSize()
-        option.features = QStyleOptionButton.Flat
+        option.icon = self.icon()
+        option.text = ""
         if self.isDown():
             option.state |= QStyle.State_Sunken
+
+    def paintEvent(self, event):
+        painter = QStylePainter(self)
+        option = QStyleOptionButton()
+        self.initStyleOption(option)
+        if self.isDown():
+            # Use default pressed flat button look
             painter.drawPrimitive(QStyle.PE_PanelButtonBevel, option)
+        elif option.state & (QStyle.State_MouseOver | QStyle.State_HasFocus):
+            flat_button_hover_background(painter, option)
 
         if not option.icon.isNull():
+            rect = option.rect.adjusted(1, 1, -1, -1)
             if option.state & QStyle.State_Active:
                 mode = (QIcon.Active if option.state & QStyle.State_MouseOver
                         else QIcon.Normal)
+            elif option.state & QStyle.State_Enabled:
+                mode = QIcon.Normal
             else:
                 mode = QIcon.Disabled
             if self.isChecked():
                 state = QIcon.On
             else:
                 state = QIcon.Off
-            option.icon.paint(painter, option.rect, Qt.AlignCenter, mode, state)
+            option.icon.paint(painter, rect, Qt.AlignCenter, mode, state)
+
+
+def flat_button_hover_background(
+        painter: QPainter, option: QStyleOption
+):
+    palette = option.palette
+    g = QLinearGradient(0, 0, 0, 1)
+    g.setCoordinateMode(QLinearGradient.ObjectBoundingMode)
+    base = palette.color(QPalette.Window)
+    base.setAlpha(200)
+    if base.value() < 127:
+        base_ = base.lighter(170)
+    else:
+        base_ = base.darker(130)
+    g.setColorAt(0, base_)
+    g.setColorAt(0.6, base)
+    g.setColorAt(1.0, base_)
+    brush = QBrush(base_)
+    cg = palette_color_group(option.state)
+    if option.state & QStyle.State_HasFocus:
+        pen = palette.color(cg, QPalette.Highlight)
+    elif option.state & QStyle.State_MouseOver:
+        pen = palette.color(cg, QPalette.Foreground)
```

Comment:
Foreground is deprecated and removed in PyQt6 use WindowText instead
```suggestion
        pen = palette.color(cg, QPalette.WindowText)
```